RUFAS/routines/animal/ration/ration_driver.py

```diff
@@ -309,64 +326,580 @@ class RationReporter:
+        if feed_item_info["type"] == "Mineral":
+            NE_growth = 0.0
+        elif feed_item_info["is_fat"] == 1:
+            NE_growth = 0.55 * ME_item
+        else:
+            NE_growth = 1.42 * ME_item - 0.174 * ME_item**2 + 0.0122 * ME_item**3 - 1.65
+        return NE_growth * kg_fed
+
+    @classmethod
+    def get_calcium(cls, kg_fed: float, feed_item_info: Dict, ration_report: Dict, body_weight: float) -> float:
+        """
+        Returns calcium supply of feed item
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Calcium supply of feed item, kg TODO: check units
+
+        """
+        if feed_item_info["type"] == "Forage":
+            dCa = 0.3
+        elif feed_item_info["type"] == "Conc":
+            dCa = 0.6
+        elif feed_item_info["type"] == "Mineral":
+            dCa = 0.95
+        else:
+            dCa = 0.0
+        calcium_item = kg_fed * feed_item_info["calcium"] * 0.01 * dCa
+        return calcium_item
+
+    @classmethod
+    def get_phosphorus(cls, kg_fed: float, feed_item_info: Dict, ration_report: Dict, body_weight: float) -> float:
+        """
+        Returns phosphorus supply of feed item
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Phosphorus supply, kg TODO: check units
+
+        """
+        if feed_item_info["type"] == "Forage":
+            dP = 0.64
+        elif feed_item_info["type"] == "Conc":
+            dP = 0.70
+        elif feed_item_info["type"] == "Mineral":
+            dP = 0.80
+        else:
+            dP = 0.0
+        return dP * feed_item_info["phosphorus"] * 0.01 * kg_fed
+
+    @classmethod
+    def get_fat(cls, kg_fed, feed_item_info, ration_report, body_weight):
+        """
+        Returns fat supply of feed item
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Fat supply of feed item, g TODO: check units
```

Comment:
```suggestion
        Returns fat supply of feed item.

        Parameters
        ----------
        kg_fed : float
            Kilograms of feed item in ration.
        feed_item_info : Dict
            Dictionary of nutrient and energy information of feed item.
        ration_report : Dict
            Dictionary of nutrient amount and concentrations.
        body_weight : float
            Animal body weight in kg.

        Returns
        -------
        float
            Fat supply of feed item, g TODO: check units.
```
Same comment on TODO here as above